app/src/main/java/org/thoughtcrime/securesms/glide/cache/EncryptedCoder.java

```diff
@@ -0,0 +1,98 @@
+package org.thoughtcrime.securesms.glide.cache;
+
+
+import androidx.annotation.NonNull;
+
+import org.signal.core.util.StreamUtil;
+import org.thoughtcrime.securesms.util.Util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.Mac;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+class EncryptedCoder {
+
+  private static byte[] MAGIC_BYTES = {(byte)0x91, (byte)0x5e, (byte)0x6d, (byte)0xb4,
+                                       (byte)0x09, (byte)0xa6, (byte)0x68, (byte)0xbe,
+                                       (byte)0xe5, (byte)0xb1, (byte)0x1b, (byte)0xd7,
+                                       (byte)0x29, (byte)0xe5, (byte)0x04, (byte)0xcc};
+
+  OutputStream createEncryptedOutputStream(@NonNull byte[] masterKey, @NonNull File file)
+      throws IOException
+  {
+    try {
+      byte[] random = Util.getSecretBytes(32);
+      Mac    mac    = Mac.getInstance("HmacSHA256");
+      mac.init(new SecretKeySpec(masterKey, "HmacSHA256"));
+
+      FileOutputStream fileOutputStream = new FileOutputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fileOutputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Cipher.init throws an exception. Ensure *fileOutputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).